build.rs

```diff
@@ -42,21 +67,38 @@ fn main() {
             .rustified_enum("nng_pipe_ev")
             .rustified_enum("nng_sockaddr_family")
             .rustified_enum("nng_zt_status")
+            .rustified_enum("nng_tls_mode")
+            .rustified_enum("nng_tls_auth_mode")
+            .rustified_enum("nng_http_status")
+            .use_core();
+
+        if cfg!(feature = "nng-compat") {
+            builder = builder.header("compat.h");
+        }
+        if cfg!(feature = "nng-supplemental") {
+            builder = builder.header("supplemental.h");
+        }
+        if cfg!(feature = "no_std") {
             // no_std support
             // https://rust-embedded.github.io/book/interoperability/c-with-rust.html#automatically-generating-the-interface
-            .ctypes_prefix("cty")
-            .use_core();
+            builder = builder.ctypes_prefix("cty")
+        }
+
+        builder
+            .generate()
+            .expect("Unable to generate bindings")
+            .write_to_file(out_path)
```

Comment:
From the [Rust docs][1]:

> In general, build scripts should not modify any files outside of `OUT_DIR`. It may seem fine on the first blush, but it does cause problems when you use such crate as a dependency, because there's an implicit invariant that sources in `.cargo/registry` should be immutable. `cargo` won't allow such scripts when packaging.

I think this means that we won't be able to publish the package if we do it this way. Even if we were able to publish, I think it would be better style to output to the correct directory and then use a use conditional compilation in the `lib.rs` to select the correct file.

[1]: https://doc.rust-lang.org/cargo/reference/build-scripts.html

Reply:
I was thinking about how to deal with crate immutability after submitting the PR.  I was going to copy bindings.rs to `OUT_DIR`, but you're right, that's probably easier.